frappe/core/doctype/file/utils.py

```diff
@@ -360,6 +362,7 @@ def attach_files_to_document(doc: "Document", event) -> None:
 			attached_to_doctype=doc.doctype,
 			attached_to_field=df.fieldname,
 			folder="Home/Attachments",
+			is_private=is_private,
```

Comment:
Although I cannot replicate this, this fix should be better handled by https://github.com/frappe/frappe/pull/27352

Reply:
Yes, you are right. I've tested my case with #27352, and it behaves correctly.